Source/WebCore/inspector/InspectorFrontendHost.cpp

```diff
@@ -421,12 +421,12 @@ String InspectorFrontendHost::platform() const
 
 String InspectorFrontendHost::platformVersionName() const
 {
-#if PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000
-    return "monterey"_s;
-#elif PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110000
-    return "big-sur"_s;
-#elif PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 101500
-    return "catalina"_s;
```

Comment:
We can't just change these and replace them with contemporary versions. There are side-effects.

The only consumer of this API is on the frontend, in `WI.Platform`: https://github.com/WebKit/WebKit/blob/6cdabf3838fde6f3f97bd2ff0f8a6027fe568a47/Source/WebInspectorUI/UserInterface/Base/Platform.js#L26-L31

It is used to set class names on the `body` element of Web Inspector UI to allow for platform-dependent styling:

```
body:is(.mac-platform.monterey, .mac-platform.big-sur)
```

Usage seen in:
- https://github.com/WebKit/WebKit/blob/main/Source/WebInspectorUI/UserInterface/Views/Variables.css
- https://github.com/WebKit/WebKit/blob/main/Source/WebInspectorUI/UserInterface/Views/Main.css
- https://github.com/WebKit/WebKit/blob/main/Source/WebInspectorUI/UserInterface/Views/TabBar.css

That likely made sense in the transition period between supported platforms to use platform-consistent styling.

But Big Sur and Catalina are platforms which no longer get the latest Web Inspector. We can remove the CSS conditions and associated code for those cases.

Monterey (and onwards) is the supported platform-dependent styling. 
Contemporary Web Inspector UI on Sequoia uses this:

```css
<body class="mac-platform monterey ...">
```

Just adding "Sequoia"  in `InspectorFrontendHost::platformVersionName()` without updating the CSS, will cause fallback to older styles or default values that change or break the current UI.

But do we need platform-dependent styling anymore? 
I'd say we can:
- use any styles guarded by `.mac-platform.monterey` as defaults
- remove any styles guarded by `.mac-platform.monterey`, `.mac-platform.big-sur` and `mac-platform.catalina`

Do you have any objections @burg, @dcrousso and @patrickangle? 

After making those changes, `InspectorFrontendHost::platformVersionName()` can be updated to include the latest platforms (for any other potential uses of the API) and remove Big Sur and Catalina (no longer supported).

Reply:
if Catalina, Big Sur, or Monterey no longer receive updates (i think the latest version that does is Ventura) then yeah we should remove all relevant code in `Source/WebInspectorUI/UserInterface` that uses `catalina`/`big-sur`/`monterey`/etc.

also, unless there is some unique styling or functionality of a particular version then i dont think we should even have anything inside `InspectorFrontendHost::platformVersionName`.  put another way, we only need to care about the platform version if the differences from the previous version are not otherwise discoverable via JS/CSS/etc. (e.g. the system UI design changes)

Reply:
> also, unless there is some unique styling or functionality of a particular version then i dont think we should even have anything inside InspectorFrontendHost::platformVersionName.

There are no other users of this API right now so it could be removed. But it is an elegant solution to expose the platform name to the frontend. Normally, we wouldn't keep unused code around. I'm inclined to keep it because it's small and focused, and makes `WI.Platform` generically useful. But I don't feel too strongly about that.

Reply:
@AreaZR Are you interested in working (in a separate patch) to remove the obsolete CSS and update the default styles with the ones guarded by `body:is(.mac-platform.monterey)`?

If not, I can take that task.

Reply:
I think you should only because I don't know where the code I would have to remove for this change is.

Reply:
I'll handle it 👍. Filed https://bugs.webkit.org/show_bug.cgi?id=281910